ethereum/test/test_beefy_light_client.js

```diff
@@ -28,29 +28,35 @@ describe("Beefy Light Client", function () {
 
     this.timeout(10 * 1000)
 
-    this.validatorAddressOne = "0xE04CC55ebEE1cBCE552f250e85c57B70B2E2625b";
-    this.validatorAddressTwo = "0x25451A4de12dcCc2D166922fA938E900fCc4ED24";
-    this.validatorsMerkleTree = createMerkleTree([this.validatorAddressOne, this.validatorAddressTwo]);
-    this.validatorsLeaf0 = this.validatorsMerkleTree.getHexLeaves()[0];
-    this.validatorsLeaf1 = this.validatorsMerkleTree.getHexLeaves()[1];
-    this.validator0PubKeyMerkleProof = this.validatorsMerkleTree.getHexProof(this.validatorsLeaf0);
-    this.validator1PubKeyMerkleProof = this.validatorsMerkleTree.getHexProof(this.validatorsLeaf1);
+    this.validatorsMerkleTree = createMerkleTree(fixture.validatorPublicKeys);
+    this.beefyLightClient = await deployBeefyLightClient(this.validatorsMerkleTree.getHexRoot(),
+      fixture.validatorPublicKeys.length);
+  });
+
+  it("encodes beefy mmr leaves correctly", async function () {
+    encodedLeaf = await this.beefyLightClient.encodeMMRLeaf(fixture.beefyMMRLeaf)
+    expect(encodedLeaf).to.eq(fixture.encodedBeefyLeaf)
+  });
 
-    this.beefyLightClient = await deployBeefyLightClient(this.validatorsMerkleTree.getHexRoot(), this.validatorsMerkleTree.getLeaves().length);
+  it("hashes beefy mmr leaves correctly", async function () {
+    hashedLeaf = await this.beefyLightClient.hashMMRLeaf(fixture.encodedBeefyLeaf)
+    expect(hashedLeaf).to.eq(fixture.hashedBeefyLeaf)
   });
 
   it("runs new signature commitment and complete signature commitment correctly", async function () {
-    const initialBitfield = await this.beefyLightClient.createInitialBitfield([0, 1], 2);
+    const initialBitfield = await this.beefyLightClient.createInitialBitfield(fixture.validatorBitfield, 2);
     expect(printBitfield(initialBitfield)).to.eq('11')
 
-    await this.beefyLightClient.newSignatureCommitment(
-      fixture.commitmentHash,
+    const commitmentHash = await this.beefyLightClient.createCommitmentHash(fixture.commitment);
+
+    const tx = await this.beefyLightClient.newSignatureCommitment(
+      commitmentHash,
       initialBitfield,
-      signatureSubstrateToEthereum(fixture.signature0),
+      fixture.signatures[0],
```

Comment:
If you do the signature conversion when creating the fixture now, shall we add a section to the README on how to update the fixture?

Reply:
https://github.com/Snowfork/polkadot-ethereum/pull/442/commits/6919266bcae1c8cdb7318e3bdf8e7a99e9f045fc